src/api_proxy/service_control/request_builder.cc

```diff
@@ -384,8 +384,16 @@ Status set_credential_id(const SupportedLabel& l, const ReportRequestInfo& info,
       api_key::ApiKeyState::VERIFIED) {
     ASSERT(!info.api_key.empty(),
            "API Key must be set, otherwise consumer would not be verified.");
-    (*labels)[l.name] = absl::StrCat(kApiKeyPrefix, info.check_response_info.api_key_uid.empty() ? info.api_key
+    if (info.report_api_key_uid) {
+      if (!info.check_response_info.error.name.empty()) {
+        (*labels)[l.name] = absl::StrCat(kApiKeyPrefix, "UNKNOWN");
+      } else {
+        (*labels)[l.name] = absl::StrCat(kApiKeyPrefix, info.check_response_info.api_key_uid.empty() ? info.api_key
                       : info.check_response_info.api_key_uid);
```

Comment:
The api_key_uid returned in CheckResponse contains the prefix "apikey:", or you have strip that before saving it under info.check_response_info?

Reply:
I don't think so. let me remove the prefix. Will add integration test later to make sure it works.

Reply:
how about now?